library/hawkular_alerts_group_trigger.py

```diff
@@ -210,15 +208,19 @@ def required_updates(self, trigger, name, severity, enabled):
     def set_group_trigger_conditions(self, group_id, conditions):
         """ Set the conditions for the group trigger
         """
-        gc = hawkular.alerts.GroupConditionsInfo()
-        for c in conditions["conditions"]:
+        firing_gc = hawkular.alerts.GroupConditionsInfo()
+        autoresolve_gc = hawkular.alerts.GroupConditionsInfo()
+        for c in conditions:
             name = c.pop("name")
             condition = hawkular.alerts.Condition(c)
-            condition.trigger_mode = conditions["trigger_mode"]
             condition.context = {'name': name}
-            gc.addCondition(condition)
+            if condition.trigger_mode == "FIRING":
+                firing_gc.addCondition(condition)
+            elif condition.trigger_mode == "AUTORESOLVE":
+                autoresolve_gc.addCondition(condition)
         try:
-            self.client.create_group_conditions(group_id, conditions["trigger_mode"], gc)
+            self.client.create_group_conditions(group_id, "FIRING", firing_gc)
+            self.client.create_group_conditions(group_id, "AUTORESOLVE", autoresolve_gc)
```

Comment:
you're right, but it does complicate.. 
maybe you can mention it as an issue  for future refactoring?